app/assets/javascripts/components/AccountMenuReact.tsx

```diff
@@ -408,11 +412,74 @@ const AccountMenu = observer(({ application, appState, closeAccountMenu }: Props
   };
 
   const downloadDataArchive = () => {
-    console.log('downloadDataArchive');
+    application.getArchiveService().downloadBackup(isBackupEncrypted);
+  };
+
+
+  const readFile = async (file: File): Promise<any> => {
+    return new Promise((resolve) => {
+      const reader = new FileReader();
+      reader.onload = (e) => {
+        try {
+          const data = JSON.parse(e.target!.result as string);
+          resolve(data);
+        } catch (e) {
+          application.alertService.alert(STRING_INVALID_IMPORT_FILE);
+        }
+      };
+      reader.readAsText(file);
+    });
+  };
+
+  const performImport = async (data: BackupFile) => {
+    setIsImportDataLoading(true);
+
+    const result = await application.importData(data);
+
+    setIsImportDataLoading(false);
+
+    if (!result) {
```

Comment:
hmm maybe a switch block would be more in place instead of if..else if...else ?

Reply:
There are different variables in each `if` condition but `switch` expects one expression inside it. Agree that this could be improved, made it more readable now.